Protocol.md

```diff
@@ -0,0 +1,177 @@
+# DLC Protocol
+
+This document describes a protocol for two parties to set up a Discreet Log Contract (DLC).
+
+## Preliminaries
+
+The following protocol assumes that both parties have already agreed on the contract terms (the possible outputs of the contract).
+Negotiation of the contract, or decisions on the underlying asset are thus not part of the exchanged information.
+
+## Description
+
+The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's public key
+1. A's inputs
+1. A's change outputs (optional)
```

Comment:
Should only be one output (at least for now that is how it is setup in Transaction.md) and I don't think there should be an option not to specify this as it is necessary for constructing the funding tx

Reply:
Thanks the `s` was a typo. The reason it is optional is that A (or B) might not need a change output. Do you think we should specify 0?

Reply:
I guess it is true that if the funding inputs are just right then there should not be a change output, but I think that this should be handled in transaction construction (i.e. removing outputs below the dust limit) rather than here as I predict that having this be optional will cause people to lose change more likely than to publish unusable utxos. I guess I think we should just have it be uniform and non-optional but that it should get dropped if it isn't needed (which is a deterministic thing since the fee rate is agreed to here)

Reply:
Based on your arguments, I wonder if we shouldn't just drop this then. Since we have the fee rate and the extended pub key with derivation scheme, we should be able to determine whether to add a change output and how to construct it if needed?

Reply:
I think that we decided that for now we will make the change SPK specified to avoid forcing derivation to be off of this DLC's key chain because funding change should not necessarily be owned by the custodian of this DLC.

Reply:
I just wanted to clarify something here. Should it be a change output (including amount and address) or just an address and then the change is computed automatically? I feel the second option is a bit easier but I just wanted to confirm.

Reply:
I agree that it should be the second option: just an address

This being because can compute change deterministically from this

Reply:
Removed the optional and changed to be just an address.